### PR TITLE
Issue #46: Clear raffle winner

### PR DESCRIPTION
**Pull request by benginerd**
The previous manual solution was to write a script against the db using user_id and raffle_id. This solution uses the user_id and looks for a raffle win occurring within the last 7 days.

### CODE REVIEW

**Comment by benginerd** [comment]
https://streamable.com/2ryedl

**Review by teknogeek — CHANGES_REQUESTED** [review]

**Comment by teknogeek** [comment]
LGTM overall, relying on the last 7 days is convenient and will save us from having to search for the message ID whenever we need to do a win reset from a modmail ticket

File: db/__init__.py [hunk 1]
```diff
@@ -290,6 +290,27 @@ def clear_win(self, raffle_message_id: int) -> None:
                 .execution_options(synchronize_session="fetch")
             )
 
+    def remove_raffle_winner(self, user_id: int, after: datetime) -> None:
+        with self.session() as sess:
+            # removes winner from raffle entry within last week, so winner can win again
+            
+            raffle_id_lookup = sess.query(RaffleEntry.raffle_id).filter(
+            RaffleEntry.user_id == user_id, RaffleEntry.winner == 1, RaffleEntry.timestamp > after).scalar()
```
[the remark this answers is not in the document]
Comment (benginerd) [reply db/__init__.py]: Good point, I updated and believe I applied this. Let me know!

**Review by teknogeek — CHANGES_REQUESTED** [review]
LGTM otherwise

File: db/__init__.py [hunk 2]
```diff
@@ -290,6 +290,39 @@ def clear_win(self, raffle_message_id: int) -> None:
...
+                select(Raffle.id)
+                .select_from(RaffleEntry)
+                .join(Raffle)
+                .where(Raffle.guild_id == guild_id)
+                .where(Raffle.ended == True)
+                .where(Raffle.raffle_type == RaffleType.normal)
+                .where(RaffleEntry.winner == True)
+                .where(RaffleEntry.user_id == user_id)
+                .where(Raffle.end_time > after)
+                .limit(1)
```
[the remark this answers is not in the document]
Comment (benginerd) [reply db/__init__.py]: Heard, I believe I applied this feedback. Originally, I had the cooldown logic do the 7 day cooldown relative to the time of day the winner originally won their raffle (datetime.now() versus datetime.now().date() ); I agree that making the cooldown only use the days and not hours etc. makes more sense and is also more intuitive for the user.

**Review by teknogeek — APPROVED** [review]

**Pull request merged**